internal/graph/storagewrapper.go

```diff
@@ -165,35 +158,51 @@ func (c *CachedDatastore) Read(
 
 	var b strings.Builder
 	b.WriteString(
-		fmt.Sprintf("%sr%s/%s", QueryCachePrefix, store, tuple.TupleKeyToString(tupleKey)),
+		storage.GetReadCacheKey(store, tuple.TupleKeyToString(tupleKey)),
 	)
-	return c.newCachedIterator(ctx, iter, b.String())
+	return c.newCachedIterator(ctx, store, iter, b.String(), storage.GetInvalidIteratorByObjectRelationCacheKey(store, tupleKey.GetObject(), tupleKey.GetRelation()))
```

Comment:
There is a strong assumption here that datastore.Read has object and relation defined.  This is true where we are currently using it.  However, can we document this AND add some form of check to ensure this is the case (i.e., throw out internal error)?  This allows us not to forget about it in the future and accidentally expand the use when we should not.